### ISSUE TITLE
"Failed to submit rendering: Failed to submit result for display"

### ISSUE THREAD

**Issue by Mafrans** [pop-os/cosmic-comp#872]
May be a duplicate of https://github.com/pop-os/cosmic-comp/issues/68.

Cosmic-comp renders cosmic-greeter fine (with some graphical glitches but otherwise perfectly functional) but when I attempt to log in I end up in a sort of boot-loop where cosmic-comp restarts itself over and over only to then die with the following error:
```
Sep 19 19:29:15 DESKTOP-859OH10.viasat.nds.com cosmic-comp[2343]: Failed to submit rendering: Failed to submit result for display
                                                                  
                                                                  Caused by:
                                                                      0: The underlying drm surface encountered an error: DRM access error: Page flip commit failed on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
                                                                      1: DRM access error: Page flip commit failed on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
                                                                      2: Permission denied (os error 13)

```
I have uploaded a journal to https://0x0.st/X3w0.txt

Neofetch output:
```
OS: Arch Linux x86_64
Host: Laptop 13 (Intel Core Ultra Series 1) A2
Kernel: 6.10.10-arch1-1
CPU: Intel Ultra 5 125H (18) @ 4.500GHz
GPU: Intel Arc Graphics
```

**Comment by Drakulix** [comment]
Well that log certainly doesn't look like a drm flip error, but like a crash due to llvmpipe being used. Which is weird, but definitely points to a configuration error on your part. You are missing GL drivers for your GPU.

**Comment by Mafrans** [comment]
Both `mesa` and `intel-media-driver` is installed.
`eglinfo -B` shows the following: https://0x0.st/X3x3.txt

**Comment by Drakulix** [comment]
Surprised it would select llvmpipe then. Could you post the output of `drm_info`?

**Comment by Mafrans** [comment]
> Could you post the output of drm_info?
Rather long, uploaded to https://0x0.st/X339.txt

**Comment by arashsm79** [comment]
Disabling SELinux fixed this for me.

**Comment by Mafrans** [comment]
It has been some time, and I thought I'd take a look at this once again. Updating cosmic to its latest version does not resolve the issue, but it does reach slightly further into the rendering process before llvmpipe once again crashes. Previously, no information was rendered before the segfault, but now the background picture is rendered correctly for about one second before crashing.

I was suspecting Intel Arc's npu drivers not being installed might be causing the issues, but while updating them did resolve one of the error messages the issue persists.

`sudo journalctl -b` output: https://0x0.st/Xgl1.txt

**Comment by Mafrans** [comment]
Closing this issue as I found the solution, which I'll try my best to explain below in case someone else has this issue in the future. Some digging and experimenting led me to [this post on the Arch forums](https://bbs.archlinux.org/viewtopic.php?id=287936) about simpledrm taking up the /dev/dri/card0 slot and causing rendering issues, apparently this happens when the simpledrm driver loads before your native gpu driver, which causes a fallback to llvmpipe (and, in the case of COSMIC, crashes). 

The solution I found was to simply configure my bootloader to blacklist the simpledrm platform initializer by adding the `initcall_blacklist=simpledrm_platform_driver_init` parameter to my blacklist, this is not the best solution as it disables simpledrm completely, but for my purposes it appears to do the trick. Would appreciate if someone with a little more Linux-knowledge could suggest a better alternative.

**Comment by Elias-Graf** [comment]
I'm getting this error only after I suspend and try to wake up the system again (to be fair I can't even put the system to sleep, since it just immediately starts itself again, not sure if that is related). I'm getting no output on my monitors, even though the computer is definitely running. I can switch to a TTY and get the login screen, switching back to the cosmic session, my monitors again receive no signal.
I get thousands upon thousands of these errors > 150k lines in a couple of minutes.

Relevant log might be:

<details>

```
Oct 02 17:15:55 perseus .cosmic-comp-wrapped[2246]: Failed to submit rendering: Failed to submit result for display

                                                    Caused by:
                                                        0: The underlying drm surface encountered an error: DRM access error: Page flip commit failed on device `Some("/dev/dri/c>
                                                        1: DRM access error: Page flip commit failed on device Some("/dev/dri/card1") (Invalid argument (os error 22))
                                                        2: Invalid argument (os error 22)
Oct 02 17:15:55 perseus wireplumber[2409]: wplua: [string "alsa.lua"]:182: attempt to concatenate a nil value (local 'node_name')
                                           stack traceback:
                                                   [string "alsa.lua"]:182: in function <[string "alsa.lua"]:175>
Oct 02 17:15:55 perseus dbus-daemon[1515]: [system] Rejected send message, 7 matched rules; type="method_call", sender=":1.67" (uid=1000 pid=2465 comm="cosmic-applet-network" la>
Oct 02 17:15:55 perseus dbus-daemon[1515]: [system] Rejected send message, 7 matched rules; type="method_call", sender=":1.68" (uid=1000 pid=2437 comm="cosmic-applet-network" la>
Oct 02 17:15:55 perseus systemd[1]: systemd-suspend.service: Deactivated successfully.
Oct 02 17:15:55 perseus systemd-sleep[142851]: Successfully thawed unit 'user.slice'.
Oct 02 17:15:55 perseus kernel: PM: suspend exit
Oct 02 17:15:55 perseus bluetoothd[1513]: Controller resume with wake event 0x0
Oct 02 17:15:55 perseus systemd-sleep[142851]: System returned from sleep operation 'suspend'.
Oct 02 17:15:55 perseus kernel: random: crng reseeded on system resumption
Oct 02 17:15:55 perseus kernel: Restarting tasks ... done.
Oct 02 17:15:55 perseus kernel: OOM killer enabled.
Oct 02 17:15:55 perseus kernel: usb usb2: root hub lost power or was reset
Oct 02 17:15:55 perseus kernel: usb usb1: root hub lost power or was reset
Oct 02 17:15:55 perseus kernel: xhci_hcd 0000:04:00.2: xHC error in resume, USBSTS 0x401, Reinit
Oct 02 17:15:55 perseus kernel: ata4.00: configured for UDMA/133
Oct 02 17:15:55 perseus kernel: sd 3:0:0:0: [sda] Starting disk
Oct 02 17:15:55 perseus kernel: ata6.00: configured for UDMA/133
Oct 02 17:15:55 perseus kernel: ata6.00: supports DRM functions and may not be fully accessible
Oct 02 17:15:55 perseus kernel: ata10.00: configured for UDMA/133
Oct 02 17:15:55 perseus kernel: sd 5:0:0:0: [sdb] Starting disk
Oct 02 17:15:55 perseus kernel: sd 9:0:0:0: [sdc] Starting disk
Oct 02 17:15:55 perseus kernel: ata6.00: supports DRM functions and may not be fully accessible
Oct 02 17:15:55 perseus kernel: ata6.00: Entering active power mode
Oct 02 17:15:55 perseus kernel: ata6: SATA link up 6.0 Gbps (SStatus 133 SControl 300)
Oct 02 17:15:55 perseus kernel: ata10.00: Entering active power mode
Oct 02 17:15:55 perseus kernel: ata4.00: Entering active power mode
Oct 02 17:15:55 perseus kernel: ata10: SATA link up 6.0 Gbps (SStatus 133 SControl 300)
Oct 02 17:15:55 perseus kernel: ata4: SATA link up 6.0 Gbps (SStatus 133 SControl 300)
Oct 02 17:15:55 perseus kernel: ata12: SATA link down (SStatus 0 SControl 300)
Oct 02 17:15:55 perseus kernel: ata11: SATA link down (SStatus 0 SControl 300)
Oct 02 17:15:55 perseus kernel: usb 7-4.1: reset high-speed USB device number 3 using xhci_hcd
Oct 02 17:15:55 perseus kernel: nvme nvme0: 32/0/0 default/read/poll queues
Oct 02 17:15:55 perseus kernel: nvme nvme0: Shutdown timeout set to 10 seconds
Oct 02 17:15:55 perseus kernel: serial 00:04: activated
Oct 02 17:15:55 perseus kernel: ACPI: PM: Waking up from system sleep state S3
```
</details>

Is this related / the same as your error @Mafrans ? I do have some logs regarding "simpledrm", but they are nowhere near the period where the errors occur.

**Comment by jokeyrhyme** [comment]
I had this happen on my Dell laptop, Arch Linux, Intel + nVidia GPU, but it was seemingly fixed after updating to alpha 3

However, alpha 2 always worked just fine on my gaming PC, Arch Linux, AMD + AMD GPU,  so I'm not sure what differences/commits are and aren't relevant

**Comment by Elias-Graf** [comment]
Probably nvidia is the difference :D

**Comment by arashsm79** [comment]
I'm still facing the same error right after trying to suspend, even after adding the following boot params:

```
initcall_blacklist=simpledrm_platform_driver_init nvidia-drm.fbdev=0
```

The error:
```
Dec 31 18:14:29 cosmicbox cosmic-comp[2355]: Failed to submit rendering: Failed to submit result for display

                                             Caused by:
                                                 0: The underlying drm surface encountered an error: DRM access error: Page flip commit failed on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
                                                 1: DRM access error: Page flip commit failed on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
                                                 2: Permission denied (os error 13)
Dec 31 18:14:29 cosmicbox cosmic-comp[2355]: Error while handling udev event.
Dec 31 18:14:29 cosmicbox cosmic-comp[2355]: Error while handling udev event.
Dec 31 18:14:29 cosmicbox cosmic-comp[2355]: Unable to clear state: DRM access error: Failed to commit on clear_state on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
```

Sometimes I also get the following right before the suspend:
```
Jan 01 18:46:13 cosmicbox cosmic-comp[73546]: New screen configuration invalid!:
                                                      AtomicRequest { mapping: PropMapping { connectors: {connector::Handle(133): {"max bpc": property::Handle(135), "underscan hborder": property::Handle(37), "CRTC_ID": property::Handle(20), "Colorspace": property::Handle(136), "underscan": property::Handle(36), "DPMS": property::Handle(2), "scaling mode": property::Handle(35), "Broadcast RGB": property::Handle(134), "EDID": property::Handle(1), "Content Protection": property::Handle(98), "non-desktop": property::Handle(6), "link-status": property::Handle(5), "TILE": property::Handle(4), "underscan vborder": property::Handle(38), "HDR_OUTPUT_METADATA": property::Handle(7), "vrr_capable": property::Handle(137), "HDCP Content Type": property::Handle(99), "subconnector": property::Handle(107)}, connector::Handle(127): {"Content Protection": property::Handle(98), "CRTC_ID": property::Handle(20), "Broadcast RGB": property::Handle(128), "non-desktop": property::Handle(6), "underscan": property::Handle(36), "HDCP Content Type": property::Handle(99), "max bpc": property::Handle(129), "Colorspace": property::Handle(130), "DPMS": property::Handle(2), "subconnector": property::Handle(107), "HDR_OUTPUT_METADATA": property::Handle(7), "link-status": property::Handle(5), "underscan vborder": property::Handle(38), "EDID": property::Handle(1), "TILE": property::Handle(4), "underscan hborder": property::Handle(37), "vrr_capable": property::Handle(131), "scaling mode": property::Handle(35)}, connector::Handle(139): {"Broadcast RGB": property::Handle(140), "underscan vborder": property::Handle(38), "vrr_capable": property::Handle(143), "HDR_OUTPUT_METADATA": property::Handle(7), "EDID": property::Handle(1), "DPMS": property::Handle(2), "link-status": property::Handle(5), "max bpc": property::Handle(141), "non-desktop": property::Handle(6), "scaling mode": property::Handle(35), "TILE": property::Handle(4), "Colorspace": property::Handle(142), "CRTC_ID": property::Handle(20), "Content Protection": property::Handle(98), "underscan hborder": property::Handle(37), "subconnector": property::Handle(107), "HDCP Content Type": property::Handle(99), "underscan": property::Handle(36)}, connector::Handle(145): {"underscan vborder": property::Handle(38), "link-status": property::Handle(5), "underscan": property::Handle(36), "vrr_capable": property::Handle(149), "CRTC_ID": property::Handle(20), "scaling mode": property::Handle(35), "Colorspace": property::Handle(148), "non-desktop": property::Handle(6), "HDR_OUTPUT_METADATA": property::Handle(7), "HDCP Content Type": property::Handle(99), "max bpc": property::Handle(147), "subconnector": property::Handle(107), "Broadcast RGB": property::Handle(146), "underscan hborder": property::Handle(37), "Content Protection": property::Handle(98), "EDID": property::Handle(1), "DPMS": property::Handle(2), "TILE": property::Handle(4)}, connector::Handle(115): {"TILE": property::Handle(4), "underscan hborder": property::Handle(37), "Content Protection": property::Handle(98), "subconnector": property::Handle(107), "EDID": property::Handle(1), "underscan": property::Handle(36), "scaling mode": property::Handle(35), "max bpc": property::Handle(117), "DPMS": property::Handle(2), "CRTC_ID": property::Handle(20), "link-status": property::Handle(5), "underscan vborder": property::Handle(38), "non-desktop": property::Handle(6), "vrr_capable": property::Handle(119), "HDR_OUTPUT_METADATA": property::Handle(7), "HDCP Content Type": property::Handle(99), "Broadcast RGB": property::Handle(116), "Colorspace": property::Handle(118)}, connector::Handle(94): {"underscan vborder": property::Handle(38), "max bpc": property::Handle(95), "EDID": property::Handle(1), "Colorspace": property::Handle(96), "HDCP Content Type": property::Handle(99), "link-status": property::Handle(5), "non-desktop": property::Handle(6), "underscan hborder": property::Handle(37), "Content Protection": property::Handle(98), "DPMS": property::Handle(2), "underscan": property::Handle(36), "HDR_OUTPUT_METADATA": property::Handle(7), "vrr_capable": property::Handle(97), "TILE": property::Handle(4), "CRTC_ID": property::Handle(20), "scaling mode": property::Handle(35)}, connector::Handle(102): {"link-status": property::Handle(5), "TILE": property::Handle(4), "EDID": property::Handle(1), "vrr_capable": property::Handle(106), "non-desktop": property::Handle(6), "DPMS": property::Handle(2), "underscan": property::Handle(36), "Content Protection": property::Handle(98), "scaling mode": property::Handle(35), "underscan hborder": property::Handle(37), "underscan vborder": property::Handle(38), "CRTC_ID": property::Handle(20), "max bpc": property::Handle(104), "Colorspace": property::Handle(105), "HDCP Content Type": property::Handle(99), "Broadcast RGB": property::Handle(103), "subconnector": property::Handle(107), "HDR_OUTPUT_METADATA": property::Handle(7)}, connector::Handle(121): {"EDID": property::Handle(1), "scaling mode": property::Handle(35), "link-status": property::Handle(5), "Broadcast RGB": property::Handle(122), "HDR_OUTPUT_METADATA": property::Handle(7), "DPMS": property::Handle(2), "Content Protection": property::Handle(98), "vrr_capable": property::Handle(125), "max bpc": property::Handle(123), "Colorspace": property::Handle(124), "TILE": property::Handle(4), "non-desktop": property::Handle(6), "underscan": property::Handle(36), "CRTC_ID": property::Handle(20), "underscan hborder": property::Handle(37), "subconnector": property::Handle(107), "HDCP Content Type": property::Handle(99), "underscan vborder": property::Handle(38)}, connector::Handle(109): {"EDID": property::Handle(1), "Colorspace": property::Handle(112), "non-desktop": property::Handle(6), "HDCP Content Type": property::Handle(99), "underscan": property::Handle(36), "underscan vborder": property::Handle(38), "max bpc": property::Handle(111), "vrr_capable": property::Handle(113), "TILE": property::Handle(4), "underscan hborder": property::Handle(37), "DPMS": property::Handle(2), "link-status": property::Handle(5), "Content Protection": property::Handle(98), "HDR_OUTPUT_METADATA": property::Handle(7), "Broadcast RGB": property::Handle(110), "scaling mode": property::Handle(35), "CRTC_ID": property::Handle(20), "subconnector": property::Handle(107)}}, crtcs: {crtc::Handle(88): {"ACTIVE": property::Handle(22), "CTM": property::Handle(27), "VRR_ENABLED": property::Handle(24), "DEGAMMA_LUT": property::Handle(25), "MODE_ID": property::Handle(23), "DEGAMMA_LUT_SIZE": property::Handle(26), "OUT_FENCE_PTR": property::Handle(19), "GAMMA_LUT": property::Handle(28), "GAMMA_LUT_SIZE": property::Handle(29)}, crtc::Handle(80): {"DEGAMMA_LUT": property::Handle(25), "MODE_ID": property::Handle(23), "VRR_ENABLED": property::Handle(24), "ACTIVE": property::Handle(22), "DEGAMMA_LUT_SIZE": property::Handle(26), "GAMMA_LUT": property::Handle(28), "CTM": property::Handle(27), "GAMMA_LUT_SIZE": property::Handle(29), "OUT_FENCE_PTR": property::Handle(19)}, crtc::Handle(84): {"ACTIVE": property::Handle(22), "DEGAMMA_LUT_SIZE": property::Handle(26), "MODE_ID": property::Handle(23), "OUT_FENCE_PTR": property::Handle(19), "GAMMA_LUT": property::Handle(28), "DEGAMMA_LUT": property::Handle(25), "CTM": property::Handle(27), "VRR_ENABLED": property::Handle(24), "GAMMA_LUT_SIZE": property::Handle(29)}, crtc::Handle(92): {"DEGAMMA_LUT": property::Handle(25), "ACTIVE": property::Handle(22), "GAMMA_LUT_SIZE": property::Handle(29), "DEGAMMA_LUT_SIZE": property::Handle(26), "VRR_ENABLED": property::Handle(24), "MODE_ID": property::Handle(23), "GAMMA_LUT": property::Handle(28), "OUT_FENCE_PTR": property::Handle(19), "CTM": property::Handle(27)}}, planes: {plane::Handle(71): {"CRTC_ID": property::Handle(20), "FB_DAMAGE_CLIPS": property::Handle(21), "rotation": property::Handle(76), "type": property::Handle(8), "SRC_X": property::Handle(9), "SRC_Y": property::Handle(10), "CRTC_W": property::Handle(15), "IN_FENCE_FD": property::Handle(18), "SRC_H": property::Handle(12), "FB_ID": property::Handle(17), "SRC_W": property::Handle(11), "CRTC_Y": property::Handle(14), "zpos": property::Handle(75), "CRTC_X": property::Handle(13), "IN_FORMATS": property::Handle(30), "pixel blend mode": property::Handle(74), "alpha": property::Handle(73), "CRTC_H": property::Handle(16)}, plane::Handle(47): {"CRTC_ID": property::Handle(20), "CRTC_X": property::Handle(13), "SRC_Y": property::Handle(10), "IN_FENCE_FD": property::Handle(18), "FB_ID": property::Handle(17), "COLOR_RANGE": property::Handle(51), "CRTC_Y": property::Handle(14), "IN_FORMATS": property::Handle(30), "COLOR_ENCODING": property::Handle(50), "CRTC_W": property::Handle(15), "SRC_H": property::Handle(12), "type": property::Handle(8), "SRC_X": property::Handle(9), "zpos": property::Handle(49), "rotation": property::Handle(52), "CRTC_H": property::Handle(16), "FB_DAMAGE_CLIPS": property::Handle(21), "SRC_W": property::Handle(11)}, plane::Handle(81): {"IN_FENCE_FD": property::Handle(18), "CRTC_ID": property::Handle(20), "CRTC_H": property::Handle(16), "IN_FORMATS": property::Handle(30), "CRTC_X": property::Handle(13), "FB_ID": property::Handle(17), "CRTC_W": property::Handle(15), "SRC_W": property::Handle(11), "SRC_X": property::Handle(9), "type": property::Handle(8), "SRC_Y": property::Handle(10), "SRC_H": property::Handle(12), "zpos": property::Handle(83), "CRTC_Y": property::Handle(14)}, plane::Handle(85): {"SRC_Y": property::Handle(10), "CRTC_ID": property::Handle(20), "SRC_W": property::Handle(11), "FB_ID": property::Handle(17), "IN_FORMATS": property::Handle(30), "CRTC_H": property::Handle(16), "SRC_X": property::Handle(9), "IN_FENCE_FD": property::Handle(18), "type": property::Handle(8), "CRTC_X": property::Handle(13), "CRTC_W": property::Handle(15), "SRC_H": property::Handle(12), "CRTC_Y": property::Handle(14), "zpos": property::Handle(87)}, plane::Handle(89): {"FB_ID": property::Handle(17), "IN_FENCE_FD": property::Handle(18), "CRTC_X": property::Handle(13), "CRTC_W": property::Handle(15), "CRTC_H": property::Handle(16), "CRTC_ID": property::Handle(20), "SRC_X": property::Handle(9), "SRC_Y": property::Handle(10), "SRC_W": property::Handle(11), "SRC_H": property::Handle(12), "IN_FORMATS": property::Handle(30), "zpos": property::Handle(91), "type": property::Handle(8), "CRTC_Y": property::Handle(14)}, plane::Handle(41): {"type": property::Handle(8), "IN_FENCE_FD": property::Handle(18), "CRTC_X": property::Handle(13), "COLOR_ENCODING": property::Handle(44), "zpos": property::Handle(43), "FB_DAMAGE_CLIPS": property::Handle(21), "CRTC_W": property::Handle(15), "SRC_H": property::Handle(12), "SRC_X": property::Handle(9), "FB_ID": property::Handle(17), "CRTC_ID": property::Handle(20), "CRTC_Y": property::Handle(14), "SRC_W": property::Handle(11), "COLOR_RANGE": property::Handle(45), "SRC_Y": property::Handle(10), "IN_FORMATS": property::Handle(30), "rotation": property::Handle(46), "CRTC_H": property::Handle(16)}, plane::Handle(77): {"CRTC_H": property::Handle(16), "SRC_X": property::Handle(9), "CRTC_ID": property::Handle(20), "SRC_Y": property::Handle(10), "zpos": property::Handle(79), "CRTC_X": property::Handle(13), "SRC_W": property::Handle(11), "SRC_H": property::Handle(12), "type": property::Handle(8), "CRTC_Y": property::Handle(14), "IN_FORMATS": property::Handle(30), "CRTC_W": property::Handle(15), "IN_FENCE_FD": property::Handle(18), "FB_ID": property::Handle(17)}, plane::Handle(53): {"CRTC_H": property::Handle(16), "SRC_H": property::Handle(12), "COLOR_RANGE": property::Handle(57), "CRTC_ID": property::Handle(20), "type": property::Handle(8), "CRTC_W": property::Handle(15), "SRC_X": property::Handle(9), "CRTC_Y": property::Handle(14), "zpos": property::Handle(55), "SRC_W": property::Handle(11), "CRTC_X": property::Handle(13), "IN_FENCE_FD": property::Handle(18), "IN_FORMATS": property::Handle(30), "FB_DAMAGE_CLIPS": property::Handle(21), "COLOR_ENCODING": property::Handle(56), "rotation": property::Handle(58), "SRC_Y": property::Handle(10), "FB_ID": property::Handle(17)}, plane::Handle(65): {"FB_ID": property::Handle(17), "IN_FORMATS": property::Handle(30), "CRTC_X": property::Handle(13), "SRC_X": property::Handle(9), "SRC_H": property::Handle(12), "pixel blend mode": property::Handle(68), "type": property::Handle(8), "SRC_Y": property::Handle(10), "zpos": property::Handle(69), "rotation": property::Handle(70), "SRC_W": property::Handle(11), "CRTC_H": property::Handle(16), "CRTC_ID": property::Handle(20), "CRTC_Y": property::Handle(14), "alpha": property::Handle(67), "FB_DAMAGE_CLIPS": property::Handle(21), "CRTC_W": property::Handle(15), "IN_FENCE_FD": property::Handle(18)}, plane::Handle(59): {"CRTC_Y": property::Handle(14), "SRC_X": property::Handle(9), "COLOR_ENCODING": property::Handle(62), "SRC_W": property::Handle(11), "CRTC_W": property::Handle(15), "zpos": property::Handle(61), "CRTC_X": property::Handle(13), "CRTC_ID": property::Handle(20), "SRC_H": property::Handle(12), "FB_ID": property::Handle(17), "IN_FORMATS": property::Handle(30), "IN_FENCE_FD": property::Handle(18), "FB_DAMAGE_CLIPS": property::Handle(21), "CRTC_H": property::Handle(16), "rotation": property::Handle(64), "COLOR_RANGE": property::Handle(63), "SRC_Y": property::Handle(10), "type": property::Handle(8)}} }, request: AtomicModeReq { objects: [59, 65, 71, 77, 80, 94], count_props_per_object: [13, 14, 14, 11, 3, 1], props: [property::Handle(9), property::Handle(10), property::Handle(11), property::Handle(12), property::Handle(13), property::Handle(14), property::Handle(15), property::Handle(16), property::Handle(17), property::Handle(18), property::Handle(20), property::Handle(21), property::Handle(64), property::Handle(9), property::Handle(10), property::Handle(11), property::Handle(12), property::Handle(13), property::Handle(14), property::Handle(15), property::Handle(16), property::Handle(17), property::Handle(18), property::Handle(20), property::Handle(21), property::Handle(67), property::Handle(70), property::Handle(9), property::Handle(10), property::Handle(11), property::Handle(12), property::Handle(13), property::Handle(14), property::Handle(15), property::Handle(16), property::Handle(17), property::Handle(18), property::Handle(20), property::Handle(21), property::Handle(73), property::Handle(76), property::Handle(9), property::Handle(10), property::Handle(11), property::Handle(12), property::Handle(13), property::Handle(14), property::Handle(15), property::Handle(16), property::Handle(17), property::Handle(18), property::Handle(20), property::Handle(22), property::Handle(23), property::Handle(24), property::Handle(20)], values: [0, 0, 167772160, 104857600, 0, 0, 2560, 1600, 166, 166, 80, 163, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 18446744073709551615, 0, 0, 65535, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 18446744073709551615, 0, 0, 65535, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 18446744073709551615, 0, 1, 165, 0, 80] } }
                                                      Permission denied (os error 13)
Jan 01 18:46:13 cosmicbox cosmic-comp[73546]: Failed to submit rendering: Failed to submit result for display
                                              
                                              Caused by:
                                                  0: The underlying drm surface encountered an error: Atomic Test failed for new properties on crtc (crtc::Handle(80))
                                                  1: Atomic Test failed for new properties on crtc (crtc::Handle(80))

Jan 01 18:46:13 cosmicbox cosmic-comp[73546]: Error while handling udev event.
Jan 01 18:46:13 cosmicbox cosmic-comp[73546]: Unable to clear state: DRM access error: Failed to commit on clear_state on device `Some("/dev/dri/card0")` (Permission denied (os error 13))
```



cosmic-comp version:
```
cosmic-comp 0.1.0 (git commit fa88002ba41d2edec25dd7ffdee9719fbb928fc0)
December 10th
```

Nvidia driver:
```
| NVIDIA-SMI 580.119.02             Driver Version: 580.119.02     CUDA Version: 13.0     |
```

Laptop with two cards:
```
GPU 1: NVIDIA GeForce RTX 4060 Max-Q / Mobile [Discrete]
GPU 2: AMD Radeon 780M Graphics [Integrated]
```
